lib/mailgun/response.rb

```diff
@@ -10,6 +10,25 @@ class Response
     #   slightly different
     attr_accessor :body, :code
 
+    def self.from_hash(h)
+      # Create a "fake" response object with the data passed from h
+      resp = Object.new()
+
+      if h.key? :body then
+        resp.define_singleton_method(:body) {
+          h[:body]
+        }
+      end
+
+      if h.key? :code then
+        resp.define_singleton_method(:code) {
+          h[:code]
+        }
+      end
+
+      self.new resp
```

Comment:
Can I suggest using an OpenStruct here? Then _if_ there ever was a time where you wanted more than `body` and `code` methods, you'll get them for free.

```ruby
require 'ostruct`
self.new OpenStruct.new(h)
```
Whatever key values you pass into the `from_hash` method will be callable without checking if the key exists and defining a method for it if it does.